src/v2/Apps/Consign/Components/InDemandNow/ConsignTopArtists.tsx

```diff
@@ -85,11 +90,53 @@ const TopArtists: React.FC<ConsignTopArtistsQuery["response"]> = props => {
     return null
   }
 
+  // Iterate over microfunnel artworks and a) group into "chunks" of four for
+  // displaying in rows; b) average each artwork realized price; c) shuffle the
+  // display so that its a bit less static.
   const recentlySoldArtworks = chunk(
     shuffle(
-      microfunnelItems.map(
-        artwork => artwork?.artworksConnection?.edges?.[0]?.node
-      )
+      microfunnelItems.map(artwork => {
+        if (artwork?.artworksConnection?.edges?.length === 0) {
+          return null
+        }
+
+        const { edges } = artwork.artworksConnection
+
+        const realizedPriceAverage = edges.reduce(
+          (artworksWithPrice, artwork, index, artworks) => {
+            const realizedPrice = Number(
+              // Value comes back from MP formatted; strip dollars and commas
+              artwork.node.realizedPrice.replace("$", "").replace(",", "")
+            )
+
+            // Some artworks don't have a realized price, and return 0. We only
+            // want to average up non-zero artworks
+            if (realizedPrice) {
+              artworksWithPrice.push(realizedPrice)
+            }
+
+            if (index === artworks.length - 1) {
+              const totalValue = artworksWithPrice.reduce(
+                (total, price) => total + price,
+                0
+              )
+              const average = totalValue / artworksWithPrice.length
+              const averageInDollars = formatCentsToDollars(100 * average)
+              return averageInDollars
+            }
+            return artworksWithPrice
+          },
```

Comment:
This can be simplified like this:

```suggestion
  const realizedPriceAverage = formatCentsToDollars(100 * edges
    .map(artwork => Number(artwork.node.realizedPrice.replace("$", "").replace(",", "")))
    .filter(maybeArtworkPrice => !!maybeArtworkPrice)
    .reduce((avgArtworkPrice, artworkPrice, _, artworkPrices) => {
      return (avgArtworkPrice + artworkPrice) / artworkPrices.length
    }, 0
  ))
```

Reply:
Cool, tryin...

Reply:
Looks good 👍